### PR TITLE
Fix site-server target

### LINKED ISSUE #3169: `make site-server` is failing with a git error about dubious ownership of files
**What happened**: Running `make site-server` failed with the error:

```
Failed to read Git log: fatal: detected dubious ownership in repository at '/go/src/agones.dev/agones'
To add an exception for this directory, call:

    git config --global --add safe.directory /go/src/agones.dev/agones
```

**What you expected to happen**: It used to work without any complaints from git; it would be super handy if it still did as iterating on the docs locally is a whole lot faster than sending a PR to let the CI system generate the website. 

**How to reproduce it (as minimally and precisely as possible)**: `make site-server` (I'm on a linux machine)

**Anything else we need to know?**:

**Environment**:
- Agones version: development
- Kubernetes version (use `kubectl version`): _n/a_
- Cloud provider or hardware configuration: Running locally on a linux machine. 
- Install method (yaml/helm): _n/a_
- Troubleshooting guide log(s): _n/a_
- Others: _n/a_

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3169

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
oooh fancy!

@roberthbailey you dug into this somewhat, did you wanna take a look?

File: build/includes/website.mk [hunk 1]
```diff
@@ -23,12 +23,14 @@
 #
 # Website targets
 #
+UID := $(shell id -u)
```
Comment (igooch) [inline build/includes/website.mk:26]: Nice! You could also use the below variation as well.
`UID != id -u`
`GID != id -g`

**Review by igooch — APPROVED** [review]
LGTM

**Review by markmandel — APPROVED** [review]

File: build/includes/website.mk (hunk 1 above)
Comment (Kalaiselvi84) [reply build/includes/website.mk:26]: I apologize for the delayed response. Your suggestion seems reasonable. May we proceed with your recommendation?

**Pull request merged**